### PR TITLE
Josearmando pushpay authentication

### PR DESCRIPTION
**Pull request by josearmandohf**
Pushpay authentication, ready for code review.

Test flow:
1. Sign in as admin
2. Go to Administration/Setup/Pushpay
3. Sign in to your Pushpay admin account
4. Authorize
5. "Pushpay authorized" page displayed.
6. Go to Administration/Setup/Settings to confirm Access and Refresh token have been saved.

For this process to work a valid Touchpoint Auth Server Endpoing should be in place, I am using NGROK locally to have an Auth Test Server for this purpose, having my local has auth server endpoint should be taken as a test constraint. Open to discuss testing scenarios if necessary.

Webconfig appsetings for Pushpay must be set also.

### CODE REVIEW

File: CmsWeb/Areas/Setup/Controllers/PushpayController.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+            return Redirect("Save?_at=" + at.access_token + "&_rt=" + at.refresh_token + "&tenantRedirect=" + redirectUrl);
+
+        }
+
+        
+        public ActionResult Save(string _at, string _rt, string tenantRedirect)
+        {
+            string idAccessToken = "PushpayAccessToken", idRefreshToken= "PushpayRefreshToken";
+            var m = DbUtil.Db.Settings.AsQueryable();
```
Comment (brondavies) [inline CmsWeb/Areas/Setup/Controllers/PushpayController.cs]: When we access `DbUtil.Db` save it into a variable and re-use the variable.  Eventually this won't be the go-to method for accessing the database context.

File: CmsWeb/Areas/Setup/Controllers/PushpayController.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        { return View();  }
+
+        public async Task<AccessToken> AuthorizationCodeCallback()
+        {
+            AccessToken _accessToken;
+            // received authorization code from authorization server
+            string[] codes = Request.Params.GetValues("code");
+            var authorizationCode = "";
+            if (codes.Length > 0)
+                authorizationCode = codes[0];
```
Comment (brondavies) [inline CmsWeb/Areas/Setup/Controllers/PushpayController.cs]: You can just use `var authorizationCode = Request["code"];`

File: CmsWeb/Areas/Setup/Views/Pushpay/Finish.cshtml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    <div class="col-lg-9">
+        <div class="box box-responsive">
+
+            <div class="box-content">
+                Thank you, your account has been setup successfully.
+            </div>
+        </div>
+    </div>
+</div>
+<div class="modal fade" id="new-modal">
```
Comment (brondavies) [inline CmsWeb/Areas/Setup/Views/Pushpay/Finish.cshtml]: What is this modal used for?

File: CmsWeb/Common/Configuration.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
+﻿using System.Configuration;
+
+namespace CmsWeb.Common
```
Comment (brondavies) [inline CmsWeb/Common/Configuration.cs]: This is class is muy bueno! 👍

File: CmsWeb/Views/Shared/Menu/Admin.cshtml [hunk 5]
```diff
@@ -150,79 +150,80 @@
...
+    {
+        <li class="dropdown-header dropdown-sub-header">Setup</li>
+    }
+    @Helper.LiAnchorLink("Lookup Codes", "/Lookups/", admin || finance)
+    @Helper.LiAnchorLink("Promotion", "/PromotionSetup/", admin)
+    @Helper.LiAnchorLink("Settings", "/Settings/", admin)
+    @Helper.LiAnchorLink("Special Content", "/Display/", admin || design)
+    @Helper.LiAnchorLink("Twilio", "/Twilio/", admin)
+    @Helper.LiAnchorLink("Zip Codes", "/MetroZips/", admin)
+    @Helper.LiAnchorLink("Pushpay", "/Pushpay/", admin)
```
Comment (brondavies) [inline CmsWeb/Views/Shared/Menu/Admin.cshtml]: can we put this in alphabetical order?

File: CmsWeb/Common/Configuration.cs (hunk 4 above)
Comment (josearmandohf) [reply CmsWeb/Common/Configuration.cs]: Thanks!!! :)

**Comment by josearmandohf** [comment]
All changes are done.

**Pull request merged**